src/common/transactions/broadcast-transaction.ts

```diff
@@ -0,0 +1,37 @@
+import { broadcastRawTransaction } from '@stacks/transactions';
+import { Buffer } from 'buffer';
+import { logger } from '@common/logger';
+import { validateTxId } from '@common/validation/validate-tx-id';
+
+interface BroadcastTransactionOptions {
+  txRaw: string;
+  serialized: Buffer;
+  isSponsored: boolean;
+  attachment?: string;
+  networkUrl: string;
+}
+export async function broadcastTransaction(options: BroadcastTransactionOptions) {
+  const { txRaw, serialized, isSponsored, attachment, networkUrl } = options;
+  if (isSponsored) return { txRaw };
+  const response = await broadcastRawTransaction(
+    serialized,
+    `${networkUrl}/v2/transactions`,
+    attachment ? Buffer.from(attachment, 'hex') : undefined
+  );
+
+  if (typeof response === 'string') {
```

Comment:
It's just a preference but I like using guard style instead to keep the code flat and a bit more readable. Like this
```javascript
if (typeof response !== 'string') {
    logger.error('Error broadcasting raw transaction');
    const error = `${response.error} - ${response.reason}`;
    logger.error(error);
    throw new Error(error);
}
const isValidTxId = validateTxId(response);
if (!isValidTxId) {
    logger.error(`Error broadcasting raw transaction -- ${response}`);
    throw new Error(response);
}
return {
    txId: response,
    txRaw,
};
```

Reply:
This code is copied, but I completely agree. The early return pattern is much cleaner, and prevents nesting 👌🏼